powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/ActivationServiceBehavior.java

```diff
@@ -1336,13 +1338,25 @@ public CreateActivationResponse createActivation(CreateActivationRequest request
                         protocolVersion);
             }
 
-            final String masterPrivateKeyBase64 = masterKeyPairEntity.getMasterKeyPrivateBase64();
-            final PrivateKey privateKey = keyConvertor.convertBytesToPrivateKey(Base64.getDecoder().decode(masterPrivateKeyBase64));
+            final PrivateKey privateKey;
+            if (temporaryKeyId == null) {
```

Comment:
The could would be much more readable with inverted logic `if (temporaryKeyId != null)`.

Reply:
How? We need to specifically handle both branches, I start with the one with more business logic. 🤔

Reply:
The else branch is far away, so it is better to keep the shorter code next to the if branch for the first time reader. But feel free to ignore this comment.